discovery-infra/tests/base_test.py

```diff
@@ -90,12 +87,45 @@ def assert_string_length(string, expected_len):
         assert len(string) == expected_len, "Expected len string of: " + str(expected_len) + \
                                             " rather than: " + str(len(string)) + " String value: " + string
 
-    @staticmethod
-    def collect_test_logs(cluster, api_client, test):
+    def collect_test_logs(self, cluster, api_client, test, nodes):
         log_dir_name = f"{env_variables['log_folder']}/{test.name}"
         with suppress(ApiException):
             cluster_details = json.loads(json.dumps(cluster.get_details().to_dict(), sort_keys=True, default=str))
             download_logs(api_client, cluster_details, log_dir_name, test.result_call.failed)
+        if test.result_call.failed:
+            self._collect_virsh_logs(nodes, log_dir_name)
+
+    @staticmethod
+    def _collect_virsh_logs(nodes, log_dir_name):
+        logging.info('---TEARDOWN --- Collecting virsh logs\n')
```

Comment:
This is a utility, it wouldn't necessary be called on teardown. I suggest to remove it

Reply:
It is necessarily called in teardown, and in specific order (we make sure nodes are not teadwon before we capture all virsh data)
I dont see other usage than logs debugging.

Reply:
Nevertheless, this function is a utility, maybe it will become useful in the future.
IMO it shouldn't show `TEARDOWN` in the logs. The calling function already mentioned it's a teardown related
```
logging.info(f'--- TEARDOWN --- Collecting Logs for test: {request.node.name}\n')
```